### PR TITLE
PLT-105: Adding tenant context for AWS cloud account.

### PR DESCRIPTION
**Pull request by nikchern**

### CODE REVIEW

**Review by TylerGillson — APPROVED** [review]

**Comment by karl-cardenas-coding** [comment]
@nikchern @TylerGillson if we are adding new attributes to a resource. Can we please update the respective doc file as well. In this case, that would be this [file](https://github.com/spectrocloud/terraform-provider-spectrocloud/blob/main/docs/resources/cloudaccount_aws.md).

**Comment by TylerGillson** [comment]
> @nikchern @TylerGillson if we are adding new attributes to a resource. Can we please update the respective doc file as well? In this case, that would be this [file](https://github.com/spectrocloud/terraform-provider-spectrocloud/blob/main/docs/resources/cloudaccount_aws.md). @ritawatson and I are also available resources to help with reviewing doc changes.

@karl-cardenas-coding we should be auto-generating them. @nikchern let's put `go generate` in the pre-commit hooks.

However, it is currently failing for me with the following error:
```
rendering website for provider "terraform-provider-spectrocloud"
copying any existing content to tmp dir
exporting schema from Terraform
compiling provider "spectrocloud"
Error executing command: unable to generate website: unexpected error: invalid checksum: no checksum found in: https://releases.hashicorp.com/terraform/0.13.2/terraform_0.13.2_SHA256SUMS

exit status 1
main.go:20: running "go": exit status 1
```

**Comment by nikchern** [comment]
> > @nikchern @TylerGillson if we are adding new attributes to a resource. Can we please update the respective doc file as well? In this case, that would be this [file](https://github.com/spectrocloud/terraform-provider-spectrocloud/blob/main/docs/resources/cloudaccount_aws.md). @ritawatson and I are also available resources to help with reviewing doc changes.
> 
> @karl-cardenas-coding we should be auto-generating them. @nikchern let's put `go generate` in the pre-commit hooks.
> 
> However, it is currently failing for me with the following error:
> 
> ```shell
> rendering website for provider "terraform-provider-spectrocloud"
> copying any existing content to tmp dir
> exporting schema from Terraform
> compiling provider "spectrocloud"
> Error executing command: unable to generate website: unexpected error: invalid checksum: no checksum found in: https://releases.hashicorp.com/terraform/0.13.2/terraform_0.13.2_SHA256SUMS
> 
> exit status 1
> main.go:20: running "go": exit status 1
> ```

go generate should run on a github, not locally in my opinion as it would pick up tfvars, passwords etc.

**Comment by karl-cardenas-coding** [comment]
@nikchern @TylerGillson if we are gonna go with `go generate` don't we need the `description` attribute in the schema map? That's probably why the doc page in the registry has empty values for most attributes.

**Comment by TylerGillson** [comment]
@nikchern `go generate` can be run locally. I just fixed it and it doesn't put any sensitive info into the docs.

**Comment by karl-cardenas-coding** [comment]
@TylerGillson my team can help with that and it's something I intended to devote time in the future to address. But I think starting to add descriptions moving forward will help a lot 😄

File: spectrocloud/resource_cloud_account_aws.go [hunk 1]
```diff
@@ -25,6 +25,12 @@ func resourceCloudAccountAws() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"context": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "project",
```
Comment (karl-cardenas-coding) [inline spectrocloud/resource_cloud_account_aws.go:31]:
```suggestion
				Default:      "project",
				Description: "The scope of what the AWS account is available to in Palette. Valid options are either `project` or `tenant.` Defaults to `project`." 
```

I may have this wrong so feel free to correct me gentlemen 😄

**Comment by nikchern** [comment]
> @nikchern @TylerGillson if we are gonna go with `go generate` don't we need the `description` attribute in the schema map? That's probably why the doc page in the registry has empty values for most attributes.

We need description of course, it's just was not there historically so we have to start adding it.

File: spectrocloud/resource_cloud_account_aws.go (hunk 1 above)
Comment (nikchern) [reply spectrocloud/resource_cloud_account_aws.go:31]: Looks good to me! Thanks for the example.

**Pull request merged**